gov/src/governance.rs

```diff
@@ -0,0 +1,80 @@
+use std::sync::Arc;
+
+use ethers::contract::ContractError;
+use ethers::prelude::{signer::SignerMiddlewareError, Http, Middleware, ProviderError};
+use ethers::types::{Address, U256};
+use ethers::{
+    abi::Abi,
+    contract::{AbiError, Contract},
+    prelude::builders::ContractCall,
+    providers::Provider,
+};
+
+use std::str::FromStr;
+
+use rad_common::ethereum;
+
+const RADICLE_GOVERNANCE_ADDRESS: &str = "0x690e775361AD66D1c4A25d89da9fCd639F5198eD";
+const PUBLIC_RESOLVER_ABI: &str =
+    include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/abis/Governance.json"));
+
+pub struct Governance<M> {
+    contract: Contract<M>,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error<M: Middleware> {
+    #[error(transparent)]
+    Provider(#[from] ProviderError),
+    #[error(transparent)]
+    Contract(#[from] ContractError<M>),
+    #[error(transparent)]
+    Abi(#[from] ethers::abi::Error),
+    #[error(transparent)]
+    SignerMiddleware(#[from] SignerMiddlewareError<Provider<Http>, ethereum::Wallet>),
+}
+
+type Proposal = (Address, U256, U256, U256, U256, U256, bool, bool);
```

Comment:
This doesn't look right, ie. I think it'll fail when you `get_proposal()`: https://github.com/radicle-dev/radicle-contracts/blob/master/contracts/Governance/Governor.sol#L73

Reply:
It's only returning primitive types, arrays and mappings are missing. You can run it against a proposal id and check locally.